From 4dcc150500a9a75c2bfca53df222d88171dbdf81 Mon Sep 17 00:00:00 2001
From: Steven Morgan <stevmorg@cisco.com>
Date: Wed, 15 Nov 2017 10:46:39 -0500
Subject: bb11943 - buffer check for mew packed files.

Patch-Name: bb11943-buffer-check-for-mew-packed-files.patch
---
 libclamav/mew.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libclamav/mew.c b/libclamav/mew.c
index 14d2bc2..e5fb1f4 100644
--- a/libclamav/mew.c
+++ b/libclamav/mew.c
@@ -424,7 +424,7 @@ int mew_lzma(char *orgsource, const char *buf, uint32_t size_sum, uint32_t vma,
 		loc_edi = 1;
 		var14 = var10 = var24 = 1;
 
-                if(CLI_ISCONTAINED(orgsource, size_sum, var2C, 5))
+                if(!CLI_ISCONTAINED(orgsource, size_sum, var2C, 5))
                     return -1;
 		lzma_bswap_4861dc(&var40, var2C);
 		new_edx = 0;
